### PR TITLE
Writing test for mailbox cache and get all mailbox interactor

### PR DESCRIPTION
**Pull request by hoangdat**
@dab246 please review it

### CODE REVIEW

File: test/features/caching/mailbox_cache_client_test.dart [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    HiveCacheConfig().setUp(cachePath: Directory.current.path);
+  });
+
+  setUp(() {
+    _mailboxCacheClient = MailboxCacheClient();
+  });
+
+  group('[delete]', () {
+    test('cache should delete item successfully when cache empty', () async {
+      await _mailboxCacheClient.deleteItem(MailboxFixtures.inboxMailbox.id.toString());
```
Comment (dab246) [inline test/features/caching/mailbox_cache_client_test.dart:24]: This is a bit absurd. There is no data in the database so you can delete it and then compare again. I think in this case you are always right to leave this out,

File: test/features/caching/mailbox_cache_client_test.dart [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      await _mailboxCacheClient.insertItem(
+        MailboxFixtures.sentMailbox.id.toString(),
+        MailboxFixtures.sentMailbox.toMailboxCache());
+
+      await _mailboxCacheClient.deleteItem(MailboxFixtures.inboxMailbox.id.toString());
+
+      final remainingItems = await _mailboxCacheClient.getAll();
+
+      expect(remainingItems.length, equals(1));
+      expect(remainingItems[0], equals(MailboxFixtures.sentMailbox.toMailboxCache()));
```
Comment (dab246) [inline test/features/caching/mailbox_cache_client_test.dart:67]: IMO It is advisable to check that `remainingItems` still contains an `inboxMailbox` that is not correct in the meaning of this case.

File: test/features/caching/mailbox_cache_client_test.dart [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+          MailboxFixtures.sentMailbox.id.toString(),
+          MailboxFixtures.sentMailbox.toMailboxCache());
+
+      await _mailboxCacheClient.deleteItem(MailboxFixtures.inboxMailbox.id.toString());
+      await _mailboxCacheClient.deleteItem(MailboxFixtures.inboxMailbox.id.toString());
+
+      final remainingItems = await _mailboxCacheClient.getAll();
+
+      expect(remainingItems.length, equals(1));
+      expect(remainingItems[0], equals(MailboxFixtures.sentMailbox.toMailboxCache()));
```
Comment (dab246) [inline test/features/caching/mailbox_cache_client_test.dart:85]: I find this case you test a bit ridiculous. You have to compare it with an exception.

File: test/features/caching/mailbox_cache_client_test.dart [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+          MailboxFixtures.sentMailbox.toMailboxCache());
+
+      await _mailboxCacheClient.updateItem(
+          MailboxFixtures.inboxMailbox.id.toString(),
+          MailboxFixtures.folder1.toMailboxCache());
+
+      final remainingItems = await _mailboxCacheClient.getAll();
+
+      expect(remainingItems.length, equals(2));
+      expect(
```
Comment (dab246) [inline test/features/caching/mailbox_cache_client_test.dart:169]: You should compare the updated value, not like that.

**Comment by hoangdat** [comment]
cherry to #123

**Pull request closed without merging**